upup/pkg/fi/cloudup/hetzner/verifier.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hetzner
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/hetznercloud/hcloud-go/hcloud"
+	"k8s.io/kops/pkg/bootstrap"
+)
+
+type HetznerVerifierOptions struct {
+}
+
+type hetznerVerifier struct {
+	opt    HetznerVerifierOptions
+	client *hcloud.Client
+}
+
+var _ bootstrap.Verifier = &hetznerVerifier{}
+
+func NewHetznerVerifier(opt *HetznerVerifierOptions) (bootstrap.Verifier, error) {
+	hcloudToken := os.Getenv("HCLOUD_TOKEN")
+	if hcloudToken == "" {
+		return nil, fmt.Errorf("%s is required", "HCLOUD_TOKEN")
+	}
+
+	opts := []hcloud.ClientOption{
+		hcloud.WithToken(hcloudToken),
+	}
+	hcloudClient := hcloud.NewClient(opts...)
+
+	return &hetznerVerifier{
+		opt:    *opt,
+		client: hcloudClient,
+	}, nil
+}
+
+func (h hetznerVerifier) VerifyToken(ctx context.Context, token string, body []byte, useInstanceIDForNodeName bool) (*bootstrap.VerifyResult, error) {
+	if !strings.HasPrefix(token, HetznerAuthenticationTokenPrefix) {
+		return nil, fmt.Errorf("incorrect authorization type")
+	}
+	token = strings.TrimPrefix(token, HetznerAuthenticationTokenPrefix)
+
+	serverID, err := strconv.Atoi(token)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert server ID %q to int: %w", token, err)
+	}
+	server, _, err := h.client.Server.GetByID(context.TODO(), serverID)
```

Comment:
Is there any good reason for creating a new `context.TODO` here instead of passing in `ctx` or building on top of that one?

Reply:
Nop, copy/paste in action :).